apps/tangle-dapp/data/restake/useDelegatorTVL.ts

```diff
@@ -0,0 +1,60 @@
+import { useObservable, useObservableState } from 'observable-hooks';
+import { of, switchMap } from 'rxjs';
+
+import type { AssetMap, DelegatorInfo } from '../../types/restake';
+import safeFormatUnits from '../../utils/safeFormatUnits';
+
+export function useDelegatorTVL(
+  delegatorInfo: DelegatorInfo | null,
+  assetMap: AssetMap,
+) {
+  const tvl$ = useObservable(
+    (input$) =>
+      input$.pipe(
+        switchMap(([delegatorInfo, assetMap]) => {
+          const delegatorTVL =
+            delegatorInfo?.delegations.reduce(
+              (acc, delegation) => {
+                const assetData = assetMap[delegation.assetId];
+                if (assetData === undefined) {
+                  return acc;
+                }
+
+                const assetPrice = assetData.priceInUsd ?? null;
+
+                if (typeof assetPrice !== 'number') {
+                  return acc;
+                }
+
+                const result = safeFormatUnits(
+                  delegation.amountBonded,
+                  assetData.decimals,
+                );
+
+                if (!result.sucess) {
+                  return acc;
+                }
+
+                const formattedAmount = Number(result.value);
+
+                acc[delegation.assetId] =
+                  (acc[delegation.assetId] || 0) + formattedAmount * assetPrice;
+
+                return acc;
+              },
+              {} as Record<string, number>,
+            ) || {};
```

Comment:
```suggestion
            ) ?? {};
```
`??` instead of `||` might be more appropriate here if it returns `| undefined`